crates/duty-tracker/src/contract_state_machine.rs

```diff
@@ -3205,6 +3276,29 @@ impl ContractSM {
     }
 }
 
+fn is_valid_assignment(
```

Comment:
rename to `should_fulfill_assignment`. Validity of the assignment is not tied to whether it was assigned to us or not.